sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -245,20 +245,29 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
                 userSpecifiedSchema = Option(updatedTable.dataSchema),
                 bucketSpec = None,
                 options = options,
-                className = fileType).resolveRelation(),
+                className = fileType).resolveRelation(needPartitionInferring = false),
               table = updatedTable)
 
```

Comment:
Can we just trim inferred partition spec here?

Reply:
Do you mean trimmed here like this?
```scala
          val updatedTable = inferIfNeeded(relation, options, fileFormat)
          val baseRelation = DataSource(
            sparkSession = sparkSession,
            paths = rootPath.toString :: Nil,
            userSpecifiedSchema = Option(updatedTable.dataSchema),
            bucketSpec = None,
            options = options,
            className = fileType).resolveRelation()
          // filter out the inferred partition col
          val trimmedOutput = StructType(baseRelation.schema.filter { field =>
            updatedTable.schema.names.contains(field.name)
          }).toAttributes
          val created = LogicalRelation(baseRelation, trimmedOutput, table = updatedTable)
```
But `baseRelation`.schema is still contains partition field.

Reply:
`BaseRelation` doesn't have a copy method.

Reply:
Reverted